### PR TITLE
[imageio] Read Exif from JPEG XL images using libjxl if Exiv2 does not support it

### PR DESCRIPTION
**Pull request by victoryforce**
This PR adds support for reading Exif blobs using the native library (libjxl), this is similar to #13316 which does this for other formats. This will improve JPEG XL format support in Fedora distro where Exiv2 is built without BMFF support and therefore cannot read Exif from JPEG XL files.

### CODE REVIEW

File: src/imageio/imageio_jpegxl.c [hunk 1]
```diff
@@ -169,6 +172,31 @@ dt_imageio_retval_t dt_imageio_open_jpegxl(dt_image_t *img, const char *filename
...
+      status = JxlDecoderGetBoxType(decoder, type, JXL_FALSE);
+      if(status != JXL_DEC_SUCCESS) continue;
+
+      status = JxlDecoderGetBoxSizeRaw(decoder, &exif_size);
+      if((status != JXL_DEC_SUCCESS) || (exif_size <= 8)) continue;
+
+      if(memcmp(type, "Exif", 4) == 0)
+      {
+        // To get the Exif payload size we need to subtract 4 bytes of the box type FourCC
+        exif_size -= 4;
```
Comment (kmilos) [inline src/imageio/imageio_jpegxl.c:193]: It turns out this needs to be 8: https://github.com/libjxl/libjxl/issues/2022

Comment (victoryforce) [reply src/imageio/imageio_jpegxl.c:193]: If we subtract only 4 when the correct number would be 8, then we will read extra 4 bytes for decoding as Exif. It doesn't cause any harm, I tested the code and it works correctly. If we keep it as it is now, nothing bad will happen.

You wrote "either the API documentation needs to be updated, or the implementation needs to change to match" and I still don't see what decision will be made by the libjxl project (they only set a milestone). If the implementation is changed to match the documentation (4 must be subtracted) and we subtract 8, we will not get the last 4 bytes of Exif data.

So, despite the possible incorrectness of this value, we should keep 4 for now.

Comment (kmilos) [reply src/imageio/imageio_jpegxl.c:193]: Ok, agreed there is no side effect. But please add a comment then and a link to the libjxl issue to revisit this perhaps.

Comment (victoryforce) [reply src/imageio/imageio_jpegxl.c:193]: > But please add a comment then and a link to the libjxl issue to revisit this perhaps.

Good suggestion, I'll do.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Comment by TurboGit** [comment]
Entry for the release notes needed please. Thanks.

**Comment by kmilos** [comment]
The title is already close enough, and since we added other formats as well maybe this will do?

"Read Exif metadata from AVIF, HEIC and JPEG XL images using native libraries if Exiv2 does not support it"

**Comment by victoryforce** [comment]
@TurboGit I agree with @kmilos. It is best to make one line for all formats, without dividing by PR.

**Pull request merged**